src/mds/MDSRank.cc

```diff
@@ -2834,6 +2841,7 @@ void MDSRankDispatcher::dump_sessions(const SessionFilter &filter, Formatter *f)
       continue;
     }
 
+    s->cap_dump = cap_dump;
     f->dump_object("session", *s);
```

Comment:
specialize `Session::dump` to have another method with the `cap_dump` option. Don't save the flag in the `Session` object.